core/shared/src/main/scala/org/http4s/Entity.scala

```diff
@@ -32,13 +32,24 @@ sealed trait Entity[+F[_]] {
 }
 
 object Entity {
-  def apply[F[_]](body: EntityBody[F], length: Option[Long] = None): Entity[F] =
+
+  @deprecated("Use Entity.stream instead", "1.0.0-M38")
+  def apply[F[_]](body: Stream[F, Byte], length: Option[Long] = None): Entity[F] =
+    stream(body, length)
+
+  def stream[F[_]](body: Stream[F, Byte], length: Option[Long] = None): Entity[F] =
     Default(body, length)
 
   // The type parameter aids type inference in Message constructors.
   def empty[F[_]]: Entity[F] = Empty
   def strict(bytes: ByteVector): Entity[Pure] = Strict(bytes)
 
+  /** Generates a Strict entity with the string encoded the given charset,
+    * which is UTF_8 by default.
+    */
+  def string(str: String, charset: Charset = Charset.`UTF-8`): Entity[Pure] =
```

Comment:
The default charset argument in the Java standard library has been a significant source of bugs for a couple decades.  I would make a `utf8String`.  It could delegate to this without the default argument.